plugins/PRAplugin/src/MCSSolver.py

```diff
@@ -0,0 +1,146 @@
+      elif child.tag == 'variables':
+        variables = [str(var.strip()) for var in child.text.split(",")]
+      elif child.tag == 'map':
+        container.mapping[child.get('var')]      = child.text.strip()
+        container.invMapping[child.text.strip()] = child.get('var')
+      else:
+        raise IOError("MCSSolver: xml node " + str(child.tag) + " is not allowed")
+
+  def initialize(self, container, runInfoDict, inputFiles):
+    """
+      Method to initialize this plugin
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, runInfoDict, dict, dictionary containing all the RunInfo parameters (XML node <RunInfo>)
+      @ In, inputFiles, list, list of input files (if any)
+      @ Out, None
+    """
+    pass
+
+  def createNewInput(self, container, inputs, samplerType, **Kwargs):
+    """
+      This function has been added for this model in order to be able to create a FTstructure from multiple files
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, inputs, list, the inputs (list) to start from to generate the new one
+      @ In, samplerType, string, is the type of sampler that is calling to generate a new input
+      @ In, **kwargs, dict,  is a dictionary that contains the information coming from the sampler,
+           a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
+      @ Out, ([(inputDict)],copy.deepcopy(kwargs)), tuple, return the new input in a tuple form
+    """
+    if len(inputs) > 1:
+      raise IOError("MCSSolver: More than one file has been passed to the MCS solver")
+
+    mcsIDs, probability, mcsList, beList = mcsReader(inputs[0])
+
+    self.topEventTerms = {}
+
+    # mcsList is supposed to be a list of lists
+    # E.g., if the MCS are ABC CD and AE --> MCS1=['A','B','C'], MCS2=['D','C'], MCS3=['A','E']
+    #       then mcsList = [MCS1,MCS2,MCS3] =
+    #                    = [['A', 'B', 'C'], ['D', 'C'], ['A', 'E']]
+    # Top event should be:   ABC + CD + AE +
+    #                      - ABCD - ABCE - ACDE
+    #                      + ABCDE
+
+    for order in range(1,self.solverOrder+1):
+      self.topEventTerms[order]=[]
+      terms = list(itertools.combinations(mcsList,order))
+      # terms is a list of tuples
+      # E.g., for order=2: [ (['A', 'B', 'C'], ['D', 'C']),
+      #                      (['A', 'B', 'C'], ['A', 'E']),
+      #                      (['D', 'C'], ['A', 'E']) ]
+      self.topEventTerms[order] = []
+      for term in terms:
+        basicEventCombined = set(itertools.chain(*term))
+        # E.g. if                 term=(['A', 'B', 'C'], ['D', 'C'])
+        #      then basicEventCombined={'A', 'B', 'D', 'C'}
+        if basicEventCombined not in self.topEventTerms[order]:
+          self.topEventTerms[order].append(list(basicEventCombined))
+    return Kwargs
```

Comment:
the return is not the same as listed in the docstring

Reply:
fixed